src/Components/RideInfo/Components/RideProperty.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from "react";
+import { StyleSheet, Text, View } from "react-native";
+import withTheme, { ThemeProps } from "src/Themes/withTheme";
+
+type Style = ReturnType<typeof getStyles>;
+
+interface Props extends ThemeProps<Style> {
+  label: React.ReactNode;
+  children: React.ReactNode;
+}
+
+export const PureRideProperty = ({ label, children, styles }: Props) => (
+  <View style={styles.propertyContainer}>
+    <Text style={styles.label}>{label}</Text>
+    {typeof children === "string" ? (
```

Comment:
A very inderect way of switching between two different container styles: view and text.

Many react `i18n` libraties are always rendering as `<span>` element in web environment and in `<Text>` element. And in this check will always fall in this scenario.

In react world you can not be sure, that text only children will be typeof string, because a lot of consumers does not know how does your library makes assumptions about the content type and they will pass `text in element`, because their `i18n` lib does so by default.